### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by spk**
Hi this add initial error management let me know what you think, cheers

### CODE REVIEW

**Comment by messense** [comment]
Rebase needed.

File: src/model/errors.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+use std::fmt;
+
+#[derive(Debug)]
+pub struct RobotparserError {
```
Comment (messense) [inline src/model/errors.rs]: I'd call it `Error`, and it needs to implement `std::error::Error`

File: src/model/errors.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    pub kind: ErrorKind,
+}
+
+#[derive(Debug)]
+pub enum ErrorKind {
+    Url(url::ParseError),
+    HttpClient(reqwest::Error),
```
Comment (messense) [inline src/model/errors.rs]: `HttpClient` => `Http`

Comment (spk) [reply src/model/errors.rs]: Done in df30fe6701ba4b6df25660ded4d0161026cc37f6

File: src/model/errors.rs (hunk 1 above)
Comment (spk) [reply src/model/errors.rs]: I've implemented first but I saw that the Error description is deprecated cf https://github.com/rust-lang/rust/pull/66919 do you think we should still implement it ?

Comment (messense) [reply src/model/errors.rs]: I think you can just add a blank impl for it

Comment (spk) [reply src/model/errors.rs]: Ok thanks its done in b2525b5a

**Pull request merged**